docs/design.md

```diff
@@ -220,6 +220,40 @@ and during development, as well as any desired safety margin on the
 manufacturer's specified number of erase cycles. In general, using a ratio that
 allows hundreds to thousands of field upgrades in production is recommended.
 
+### [Swap without using scratch](#image-swap-no-scratch)
+
+This algorithm is an alternative to the swap using the scratch algorithm.
```

Comment:
I think it should be written `... to the swap using scratch algorithm.` (no **the**), not sure @FrancescoSer should know.

Reply:
You are actually right in pointing this out, but to avoid any misreading then it should be 
```suggestion
This algorithm is an alternative to the swap-using-scratch algorithm.
```